### PR TITLE
Summary table

### PR DESCRIPTION
**Pull request by danilodileo**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/metatdenovo/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/metatdenovo _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/local/collect_featurecounts.nf [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        path inputfiles
+
+    output:
+        path "*.tsv.gz"    , emit: counts
+        path "versions.yml", emit: versions
+
+    script:
+    def software = getSoftwareName(task.process)
+    def args     = task.ext.args ?: ''
+
```
Comment (emnilsson) [inline modules/local/collect_featurecounts.nf]: This row contains trailing white spaces, according to github actions (EditorConfig).

File: modules/local/collect_stats.nf [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+process COLLECT_STATS {
+    label 'process_low'
+
+    conda (params.enable_conda ? "conda-forge::r-tidyverse=1.3.1 conda-forge::r-data.table=1.14.0 conda-forge::r-dtplyr=1.1.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-508c9bc5e929a77a9708902b1deca248c0c84689:0bb5bee2557136d28549f41d3faa08485e967aa1-0' :
+        'quay.io/biocontainers/mulled-v2-508c9bc5e929a77a9708902b1deca248c0c84689:0bb5bee2557136d28549f41d3faa08485e967aa1-0' } "
+
+    input:
+
```
Comment (emnilsson) [inline modules/local/collect_stats.nf]: According to github actions (EditorConfig), there's a lot of trailing white spaces/wrong left paddings:

```
    16: Trailing whitespace
	31: Trailing whitespace
	44: Wrong amount of left-padding spaces(want multiple of 4)
	45: Wrong amount of left-padding spaces(want multiple of 4)
	46: Wrong amount of left-padding spaces(want multiple of 4)
	47: Wrong amount of left-padding spaces(want multiple of 4)
	48: Wrong amount of left-padding spaces(want multiple of 4)
	49: Wrong amount of left-padding spaces(want multiple of 4)
	50: Trailing whitespace
	50: Wrong amount of left-padding spaces(want multiple of 4)
	51: Wrong amount of left-padding spaces(want multiple of 4)
	52: Wrong amount of left-padding spaces(want multiple of 4)
	53: Wrong amount of left-padding spaces(want multiple of 4)
	54: Wrong amount of left-padding spaces(want multiple of 4)
	77: Trailing whitespace
	89: Trailing whitespace
	94: Trailing whitespace
```

File: modules/local/functions.nf [hunk 3]
```diff
@@ -0,0 +1,79 @@
+//
```
Comment (emnilsson) [inline modules/local/functions.nf]: Why is this one recreated? It shouldn't be needed after the new module syntax (that was incorporated with template 2.3).

File: workflows/metatdenovo.nf [hunk 4]
```diff
@@ -236,6 +237,26 @@ workflow METATDENOVO {
 
     FEATURECOUNTS_CDS ( ch_featurecounts)
     ch_versions       = ch_versions.mix(FEATURECOUNTS_CDS.out.versions)
+    //
+    // MODULE: Collect featurecounts output counts in one table
+    //
+
```
Comment (emnilsson) [inline workflows/metatdenovo.nf]: This row contains trailing white spaces, according to github actions (EditorConfig).

File: workflows/metatdenovo.nf [hunk 5]
```diff
@@ -236,6 +237,26 @@ workflow METATDENOVO {
...
     FEATURECOUNTS_CDS ( ch_featurecounts)
     ch_versions       = ch_versions.mix(FEATURECOUNTS_CDS.out.versions)
+    //
+    // MODULE: Collect featurecounts output counts in one table
+    //
+    
+    COLLECT_FEATURECOUNTS ( FEATURECOUNTS_CDS.out.counts.map { it[1] })
+    ch_cds_counts = COLLECT_FEATURECOUNTS.out.counts
+    ch_versions = ch_versions.mix(COLLECT_FEATURECOUNTS.out.versions)
+
```
Comment (emnilsson) [inline workflows/metatdenovo.nf]: This row contains trailing white spaces, according to github actions (EditorConfig).

**Review by emnilsson — CHANGES_REQUESTED** [review]
Just some small fixes, and a concern about the `functions.nf`. But there's one thing I'd like to confirm before approving, when I run e.g. `-profile test`, I get the following `overall_stats.tsv`:

```
sample  n_trimmed       idxs_n_mapped   idxs_n_unmapped n_feature_count
test_minigut_sample2_T1 24993   11109   4909    NA
test_minigut_sample2_T1task.sort.bam    NA      NA      NA      5446
test_minigut_T1 50000   99830   156     NA
```

Is that expected? If nothing else, I'd expect there only to be two samples, and maybe some value for featurecounts for both of them. And what's the `idxs_`?

File: modules/local/collect_stats.nf (hunk 2 above)
Comment (danilodileo) [reply modules/local/collect_stats.nf]: I should have fixed them!

**Review by emnilsson — APPROVED** [review]
LGTM

**Pull request merged**